python/pyiceberg/io/pyarrow.py

```diff
@@ -122,6 +126,12 @@
 ONE_MEGABYTE = 1024 * 1024
 BUFFER_SIZE = "buffer-size"
 ICEBERG_SCHEMA = b"iceberg.schema"
+FIELD_ID = "field_id"
+FIELD_DOC = "doc"
+PYARROW_FIELD_ID_KEYS = [b"PARQUET:field_id", b"field_id"]
```

Comment:
Why does this support `field_id` without a prefix?

Reply:
I added support for the non-prefixed `field_id` to maintain consistency with the `schema_to_pyarrow` visitor, which uses `field_id` as the metadata key. This support may be helpful in the future if we use this visitor to get the `pa.Schema` for writing data files. Additionally, it allows us to perform round-trip tests by combining `schema_to_pyarrow` and `pyarrow_to_schema`.

The `pyarrow_to_schema` visitor first looks for `PARQUET:field_id` and then falls back to `field_id`, so the behavior is as expected. However, if you still think the non-prefixed `field_id` should not be supported, please let me know.

Reply:
Sounds reasonable.